src/pom.xml

```diff
@@ -1506,45 +1532,45 @@
   </dependencies>
 
   <repositories>
+    <!-- Please do not add additional repositories as it slows the build down for everyone -->
     <repository>
       <releases>
         <enabled>true</enabled>
       </releases>
-      <!-- contains release (including third-party-dependences)               -->
-      <!-- Restlet maven Repository (http://maven.restlet.org)                             -->
-      <!-- ucar (https://artifacts.unidata.ucar.edu/content/repositories/unidata-releases) -->
       <snapshots>
         <enabled>false</enabled>
       </snapshots>
       <id>osgeo-releases</id>
       <name>OSGeo Nexus Release Repository</name>
       <url>https://repo.osgeo.org/repository/release/</url>
+      <!-- Access to OSGeo release repos:                                                  -->
+      <!-- geotools-releases                                                               -->
+      <!-- Geoserver-releases                                                              -->
+      <!-- geonetwork-releases                                                             -->
+      <!--                                                                                 -->
+      <!-- Cache of third-party repos (filtered by package):                               -->
+      <!-- Restlet maven Repository (http://maven.restlet.org)                             -->
+      <!-- ucar (https://artifacts.unidata.ucar.edu/content/repositories/unidata-releases) -->
+      <!-- geosolutions (https://maven.geo-solutions.it/)                                  -->
+      <!--                                                                                 -->
+      <!-- How to use:                                                                     -->
+      <!-- https://wiki.osgeo.org/wiki/SAC:Repo                                            -->
+      <!-- https://docs.geotools.org/latest/developer/procedures/add.html                  -->
     </repository>
 
     <repository>
       <releases>
         <enabled>false</enabled>
       </releases>
-      <!-- contains snapshots               -->
       <snapshots>
         <enabled>true</enabled>
       </snapshots>
       <id>osgeo-snapshots</id>
       <name>OSGeo Nexus Snapshot Repository</name>
       <url>https://repo.osgeo.org/repository/snapshot/</url>
+      <!-- Access to OSGeo snapshots repositories                                          -->
     </repository>
 
-    <repository>
-      <releases>
-        <enabled>true</enabled>
-      </releases>
-      <snapshots>
```

Comment:
You should get most of the speedup you're looking for by just disabling snapshot support. Operationally, it would be better for us, as only a few of the front-facing developers have access to the OSGeo infrastructure for deploy (I believe only me and Daniele) while others do not even have commit access (let alone the rest), but occasionally we have to deploy a jar for quick security fixes or to get a fix on an otherwise dead dependency (e.g., jgridshift).

Reply:
You can give more people access to upload, they just need their osgeo id. 

Still the existing caching rules should be sufficient? And I left a profile available (in case there is a community module with dependencies I do not know about).